pandas/core/generic.py

```diff
@@ -5906,6 +5906,95 @@ def infer_objects(self: FrameOrSeries) -> FrameOrSeries:
             )
         ).__finalize__(self)
 
+    def as_nullable_dtypes(
+        self: FrameOrSeries, keep_integer: bool_t = False
+    ) -> FrameOrSeries:
+        """
+        Convert columns of DataFrame or a Series to types supporting ``pd.NA``.
```

Comment:
i am not sure why this is about NA at all. This is about converting to extension types or really, 'better' dtypes. I think this method is not named to be very descrpiptive. This doc string needs some work.

Reply:
Resolving this as discussion is below.